File: Java/src/main/java/QuestionBank/Solution001_100/Solution27.java
package QuestionBank.Solution001_100;

/**
 * @author windows
 */
public class Solution27 {
    public int removeElement(int[] nums, int val) {
        int left = 0;
        for (int i = 0; i < nums.length; i++) {
           if (nums[i] != val){
               nums[left] = nums[i];
               left++;
           }
        }
        return left;
    }


    public static void main(String[] args) {
        Solution27 solution27 = new Solution27();
    }
}

File: C++/src/Solution1_100/Solution27.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution27.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution27::removeElement(vector<int> &nums, int val) {
    int i = 0;
    for (int j = 0; j < nums.size(); ++j) {
        if (nums[j] != val) {
            nums[i] = nums[j];
            i++;
        }
    }
    return i;
}
